### PR TITLE
Improve parsing & formatting of help page descriptions

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request resolves various issues that caused help page argument descriptions to not be properly parsed, leading to malformed outputs like:

```
--label (required): A unique.
```

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally and run `make install`.

### Unit Testing

```
make testunit
```

### Manual Testing

1. In your linode-cli project directory, create a new `dump_descriptions.py` file with the following contents:

```python
from linodecli import CLI
from rich.console import Console

from linodecli.baked.request import OpenAPIRequestArg

console = Console()

cli = CLI("0.0.0.dev", "https://localhost", skip_config=True)

cli.load_baked()

for op_name, op in cli.ops.items():
    for action_name, action in op.items():
        console.print(f"[green b]{op_name} {action_name}[/green b]: {action.summary}")

        if action.request is None:
            console.print()
            continue

        for arg in action.request.attrs:
            if not isinstance(arg, OpenAPIRequestArg):
                continue

            console.print(f"\t[blue b]--{arg.name}[/blue b]: {arg.description_rich}")

        console.print()
```

2. Run `python3 dump_descriptions.py`.
3. Manually review the output and ensure all descriptions are readable and do not have any formatting issues.

### CODE REVIEW

File: linodecli/baked/operation.py [hunk 1]
```diff
@@ -356,8 +357,12 @@ def __init__(
             self.action_aliases = {}
             self.action = action
 
-        self.summary = operation.summary
-        self.description = operation.description.split(".")[0]
+        # Ensure the summary has punctuation
+        self.summary = operation.summary.rstrip(".") + "."
+
+        self.description_rich, self.description = simplify_description(
```
Comment (lgarber-akamai) [inline linodecli/baked/operation.py:363]: We don't consume `OpenAPIOperation(...).description` currently but I figured it'd be good to future-proof 🙂

File: linodecli/baked/parsing.py [hunk 2]
```diff
@@ -5,15 +5,15 @@
 import functools
```
Comment (lgarber-akamai) [inline linodecli/baked/parsing.py:5]: The changes in this file originated in #639, which provides a bit more context for why these changes were made.

**Review by ezilber-akamai — APPROVED** [review]
Thanks for fixing this! Unit tests are passing locally and manual test looked good too.

**Review by zliang-akamai — APPROVED** [review]
